src/posix/platform/resolver.cpp

```diff
@@ -95,16 +97,26 @@ void Resolver::LoadDnsServerListFromConf(void)
 
     while (fp.good() && std::getline(fp, line) && mUpstreamDnsServerCount < kMaxUpstreamServerCount)
     {
-        if (line.find(kNameserverItem, 0) == 0)
+        // Skip the lines that don't start with "nameserver"
+        if (line.find(kNameserverItem, 0))
         {
-            in_addr_t addr;
+            continue;
+        }
 
-            if (inet_pton(AF_INET, &line.c_str()[sizeof(kNameserverItem)], &addr) == 1)
-            {
-                LogInfo("Got nameserver #%d: %s", mUpstreamDnsServerCount, &line.c_str()[sizeof(kNameserverItem)]);
-                mUpstreamDnsServerList[mUpstreamDnsServerCount] = addr;
-                mUpstreamDnsServerCount++;
-            }
+        const char *addressString = &line.c_str()[sizeof(kNameserverItem)];
```

Comment:
Move variable declarations top top of scope (followed by a newline).
All variable declarations should be at the top of scope (per [style guide](https://github.com/openthread/openthread/blob/555454c973107ab4c8fd39e0020bf5190342a018/STYLE_GUIDE.md#white-space)).